pyanaconda/modules/storage/bootloader/systemd.py

```diff
@@ -0,0 +1,153 @@
+    # everything is stored on the ESP
+    stage2_max_end = None
+    stage2_is_valid_stage1 = True
+    stage2_required = False
+
+    #
+    # configuration
+    #
+
+    @property
+    def config_dir(self):
+        """ Full path to configuration directory. """
+        log.info("systemd.py: config_dir")
+        esp = util.execWithCapture("bootctl", [ "--print-esp-path" ],
+                                   root=conf.target.system_root)
+        return esp.strip() + self._config_dir
+
+    @property
+    def config_file(self):
+        """ Full path to configuration file. """
+        log.info("systemd.py: config_file")
+        return "%s/%s" % (self.config_dir, self._config_file)
+
+    # copy console update from grub2.py
+    def write_config_console(self, config):
+        log.info("systemd.py: write_config_console")
+        if not self.console:
+            return
+
+        console_arg = "console=%s" % self.console
+        if self.console_options:
+            console_arg += ",%s" % self.console_options
+        self.boot_args.add(console_arg)
+
+    def write_config(self):
+        log.info("systemd.py: write_config systemd start")
+        self.write_config_console(None)
+
+        # Rewrite the loader.conf
+        # For now we are just updating the timeout to actually
+        # implement the bootloader --timeout option
+        config_path = "%s%s" % (conf.target.system_root, self.config_file)
+        log.info("systemd.py: write_config systemd loader conf : %s ", config_path)
+
+        config = open(config_path, "w")
+        config.write("timeout "+ str(self.timeout) + "\n")
+        config.write("#console-mode keep\n")
+        config.close()
+
+        # update /etc/kernel/cmdline
+        # should look something like "root=UUID=45b931b7-592a-46dc-9c33-d38d5901ec29 ro resume=/dev/sda3"
+        config_path = "%s%s" % (conf.target.system_root, "/etc/kernel/cmdline")
+        log.info("systemd.py: write_config systemd commandline : %s ", config_path)
+        config = open(config_path, "w")
+        args = " ".join(list(self.boot_args._arguments))
+        log.info("systemd.py: systemd used boot args: %s ", args)
+
+        # pick up the UUID of the mounted rootfs,
+        root_uuid = util.execWithCapture("findmnt", [ "-sfn", "-oUUID", "/" ],
+                                          root=conf.target.system_root)
+        args += " root=UUID=" + root_uuid
+        config.write(args)
+        config.close()
+        # rather than creating a mess in python lets just
+        # write the options above, and run a script which will merge the
+        # boot cmdline (after stripping inst. and BOOT_) with the anaconda
+        # settings and the kernel-install recovery/etc options.
+        rc = util.execWithRedirect(
+            "/usr/sbin/updateloaderentries",
+            [" "],
+            root=conf.target.system_root
+        )
+        if rc:
+            raise BootLoaderError("failed to write boot loader configuration")
```

Comment:
I would encourage you to provide a more detailed message if you can and mark it for translations. You can do that with: `_("This string will be translated")`. This type of an error message is a bad practice and we should finally fix that.

Reply:
This one remained the same, I simply added the "More information may be in the log files stored in /tmp" which hopefully can redirect the user at any failures should they happen. These error messages are large because of the copy/paste execWithRedirect() syle i'm copying where the failure codes are being thrown. I don't think I've seen any of them actually throw when there wasn't a bug during development.

I have put the _() around them though.

Reply:
I'm going to resolve these too, because I think they have been fixed (the _() for sure).